File: app/src/main/java/com/lkms/data/model/java/MaintenanceLog.java
package com.lkms.data.model.java;

import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Đại diện cho một hàng trong bảng "MaintenanceLog".
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class MaintenanceLog {

    @SerializedName("maintenanceId")
    private Integer maintenanceId;

    // Khóa ngoại đến User
    @SerializedName("userId")
    private Integer userId;

    // Khóa ngoại đến Equipment
    @SerializedName("equipmentId")
    private Integer equipmentId;

    @SerializedName("maintenanceTime")
    private String maintenanceTime;

    @SerializedName("maintenanceType")
    private String maintenanceType;

    @SerializedName("detail")
    private String detail;
}

File: app/src/main/java/com/lkms/data/model/MaintenanceLog.kt
// Trong file: data/model/MaintenanceLog.kt
package com.lkms.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Đại diện cho một hàng trong bảng "MaintenanceLog".
 */
@Serializable
data class MaintenanceLog(
    @SerialName("maintenanceId")
    val maintenanceId: Int,

    // Khóa ngoại đến User
    @SerialName("userId")
    val userId: Int? = null,

    // Khóa ngoại đến Equipment
    @SerialName("equipmentId")
    val equipmentId: Int? = null,

    @SerialName("maintenanceTime")
    val maintenanceTime: String? = null,

    @SerialName("maintenanceType")
    val maintenanceType: String? = null,

    @SerialName("detail")
    val detail: String? = null
)